specification/dsdl/serializable_types.tex

```diff
@@ -0,0 +1,285 @@
+\section{Serializable types}\label{sec:dsdl_serializable_types}
+
+Values of the serializable type category can be exchanged between nodes over the UAVCAN network.
+This is opposed to the expression types (section \ref{sec:dsdl_expression_types}),
+instances of which can only exist while DSDL definitions are being evaluated.
+The data serialization rules are defined in the section \ref{sec:dsdl_data_serialization}.
+
+\subsection{Void types}
+
+Void types are used for padding purposes.
+As will be explained in later sections, it is desirable to align field attributes at byte boundaries;
+void types can be used to facilitate that.
+
+Void-typed field attributes are set to zero when an object is serialized and ignored when it is deserialized.
```

Comment:
Do we have to require that they are zeroed? Can't we just specify that the actual value of the void types is undefined?

Reply:
We shouldn't lift this requirement because it simplifies the extension of data types in a backward-compatible way by adding new fields in place of padding fields. Example:

```python
# Version 1.1
float64 a
void64
```

```python
# Version 1.2
float64 a
float32 b  # Messages v1.1 will be interpreted such that b = 0.0
void32
```